commands/download.go

```diff
@@ -1,14 +1,97 @@
 package commands
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
-	"github.com/jfrog/jfrog-cli-core/plugins/components"
+	"github.com/jfrog/jfrog-cli-core/artifactory/utils"
 	"github.com/jfrog/jfrog-cli-core/utils/config"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
 	"github.com/jfrog/jfrog-client-go/utils/log"
+	"io/ioutil"
+	"net/http"
+	"time"
 )
 
-func downloadSupportBundle(ctx *components.Context, rtDetails *config.ArtifactoryDetails, conf *supportBundleCommandConfiguration, response creationResponse) (string, error) {
-	tmpFile := "/tmp/foo.tmp"
-	log.Debug(fmt.Sprintf("Download Support Bundle %s from %s to %s", response.Id, rtDetails.Url, tmpFile))
+type httpGetExecutor interface {
+	SendGet(url string, followRedirect bool, httpClientsDetails *httputils.HttpClientDetails) (resp *http.Response, respBody []byte, redirectUrl string, err error)
+}
+
+func downloadSupportBundle(timeout time.Duration, retryInterval time.Duration, rtDetails *config.ArtifactoryDetails, bundleId string) (string, error) {
+	dirPath, err := fileutils.CreateTempDir()
+	if err != nil {
+		return "", err
+	}
+	tmpFile := fmt.Sprintf("%s/%s", dirPath, bundleId)
+	log.Debug(fmt.Sprintf("Download Support Bundle %s from %s to %s", bundleId, rtDetails.Url, tmpFile))
+
+	ctx, cancelCtx := context.WithTimeout(context.Background(), timeout)
+	defer cancelCtx()
+
+	manager, err := utils.CreateServiceManagerWithProgressBar(rtDetails, 1, false, nil)
+	if err != nil {
+		return "", err
+	}
+	httpClientDetails := manager.GetConfig().GetServiceDetails().CreateHttpClientDetails()
+
+	sbStatusUrl := fmt.Sprintf("%sapi/system/support/bundle/%s", rtDetails.GetUrl(), bundleId)
+	err = waitUntilSupportBundleIsReady(ctx, manager.Client(), sbStatusUrl, httpClientDetails, retryInterval)
+	if err != nil {
+		return "", err
+	}
+
+	downloadSbUrl := fmt.Sprintf("%sapi/system/support/bundle/%s/archive", rtDetails.GetUrl(), bundleId)
+	resp, body, _, err := manager.Client().SendGet(downloadSbUrl, true, &httpClientDetails)
```

Comment:
This API loads the response body in memory. We should try `ArtifactoryHttpClient.Send()` with `closeBody` flag to `false`.

Reply:
good catch